tests/target/issue-3876.rs

```diff
@@ -0,0 +1,14 @@
+fn f<S>(s: S)
+where
+    S: Send /* */,
+             //
+{
+}
+
+fn main() {
+    foo(
+        x // comment 1
+          ,
+          // comment 2
+    )
+}
```

Comment:
This is a slight improvement in that the commas are not duplicated, though the formatting doesn't feel quite right. In particular I'm confused by the indentation/alignment in the first snippet, as well as the position of the `,` in the second.

I think a big question is whether there's a valid use case for comments _before_ the comma, or whether that's just unintentional i.e.:

```rust
fn main() {
    foo(
        x, // comment 1
        // comment 2
    )
}
```
although at the moment I suspect rustfmt may want to do this, though that's largely due to a bug in default visual alignment for comments (refs #4108)
```rust
fn main() {
    foo(
        x, // comment 1
           // comment 2
    )
}
```